File: 967-numbers-with-same-consecutive-differences/967-numbers-with-same-consecutive-differences.cpp
class Solution
{
    set<int> ans;
    void backtrack(int i, int n, int k, string temp)
    {
        if (temp.size() == n)
        {
            if (temp[0] == '0') return;
            int a = 0, m = 1;
            for (int i = n-1; i >= 0; i--) {
                a += (temp[i] - '0') * m;
                m *= 10;
            }
            ans.insert(a);
            return;
        }
        if (i + k <= 9)
            backtrack(i + k, n, k, temp + to_string(i + k));
        if (i - k >= 0)
            backtrack(i - k, n, k, temp + to_string(i - k));
    }
    public:
        vector<int> numsSameConsecDiff(int n, int k)
        {
            for (int i = 0; i <= 9; i++)
            {
                backtrack(i, n, k, "");
            }
            vector <int> res;
            for (auto c: ans) res.push_back(c);
            return res;
        }
};

File: 967-numbers-with-same-consecutive-differences/967-numbers-with-same-consecutive-differences.py
class Solution:
    def numsSameConsecDiff(self, n: int, k: int) -> List[int]:
        self.ans=set()
        def recur(s):
            if k==0 and s[-1]=='0':
                return
            if len(s)==n and s[0]!='0':
                self.ans.add(int(s))
                return
            if len(s)==n+1 and s[0]=='0':
                self.ans.add(int(s))
                return
            
            x=int(s[-1])+k
            y=int(s[-1])-k
            
            if 0<=x<10:
                recur(s+str(x))
            if 0<=y<10:
                recur(s+str(y))
        for i in range(10):
            recur(str(i))
        return list(self.ans)